FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/Hydrator/CustomListingHydratorTest.php

```diff
@@ -15,6 +15,13 @@ class CustomListingHydratorTest extends TestCase
      */
     private $hydrator;
 
+    public static function setUpBeforeClass()
+    {
+        parent::setUpBeforeClass();
+        Shopware()->Container()->reset('front');
```

Comment:
- Put these kind of things only in the test class that sets them. It shouldn't be this tests job to clean up the previous state.
- Put it in a `tearDown` method in the respective test class.

Reply:
Actually I was having errors when I had kept in tearDown but it got resolved when I added these in the setUpBefore. But I will look into it further if it can be resolved in the tearDown method directly.

Reply:
That might be because you can't remove a service from the container if it is already loaded (same issue as with the Smart Did-You-Mean tests). You will have to reset the whole container i.e. `Shopware()->Container()->reset()` and explicitly load 3 services. Just check #173. Maybe you can even copy the code.